### PR TITLE
Docs: add test for website build

### PR DESCRIPTION
**Pull request by robbymilo**
**What this PR does / why we need it**:

Adds a CircleCI job that confirms the docs website can be built without errors. Added this as a requirement to merge to master.

**Special notes for your reviewer**:

Run `circleci local execute --job build-docs-website` to test a prod build of the docs website.

Success:
Screen Shot 2020-01-06 at 5 34 15 PM

Fail:
Screen Shot 2020-01-07 at 9 27 31 AM

This error reporting is also available for contributors when running `make docs` in the `docs` directory:
Screen Shot 2020-01-06 at 5 40 08 PM

### CODE REVIEW

File: docs/README.md
Comment (oddlittlebird) [inline docs/README.md]: What do you mean by "restart"? How do I do that? 

Remember that this is likely to be read by writers, so we need to make the instructions explicit and simple.

Also, do we run this in PowerShell or the cmd window?

**Review by marefr — CHANGES_REQUESTED** [review]
Shouldn't the `build-docs-website` step be added to the `build-branches-and-prs` workflow instead or also? That way you can for starters test run these changes in circleci from this branch and it will also enforce valid doc changes before merging a PR to master. Once in master I guess the github action runs to sync website branch so `build-docs-website` step should probably not run for the `deploy-master` workflow?

File: docs/README.md
Comment (robbymilo) [reply docs/README.md]: > Also, do we run this in PowerShell or the cmd window?

The command should be run in the `docs` directory.

**Review by marefr — APPROVED** [review]
CircleCI changes LGTM. Leave it up to @oddlittlebird if there's any doc-related changes left

File: docs/README.md
Comment (oddlittlebird) [reply docs/README.md]: On my Windows machine, will I need to run this in Powershell or in the CMD window?

Also, what do you mean by "restart"? How do I do that?

Remember that this is likely to be read by writers, so we need to make the instructions explicit and simple. If you make the command something they can copy/paste, that would be ideal.

Comment (robbymilo) [reply docs/README.md]: > On my Windows machine, will I need to run this in Powershell or in the CMD window?

Wherever you run your commands.

> Also, what do you mean by "restart"? How do I do that?

Stop the process and run it again.

**Pull request merged**
